src/ServiceManagerBase.sol

```diff
@@ -135,6 +136,145 @@ abstract contract ServiceManagerBase is ServiceManagerBaseStorage {
+        uint256 rightLength = right.length;
+        uint256[] memory merged = new uint256[](leftLength + rightLength);
+
+        uint256 i = 0; // Index for left array
+        uint256 j = 0; // Index for right array
+        uint256 k = 0; // Index for merged array
+
+        // Merge the two arrays into the merged array
+        while (i < leftLength && j < rightLength) {
+            if (left[i] <= right[j]) {
+                merged[k] = left[i];
+                i++;
+            } else {
+                merged[k] = right[j];
+                j++;
+            }
+            k++;
+        }
+
+        // Copy remaining elements of left, if any
+        while (i < leftLength) {
+            merged[k] = left[i];
+            i++;
+            k++;
+        }
+
+        // Copy remaining elements of right, if any
+        while (j < rightLength) {
+            merged[k] = right[j];
+            j++;
+            k++;
+        }
+
+        return merged;
+    }
+
+    function _migrateToOperatorSets() internal {
+        // Initiate the migration process 
+        _avsDirectory.becomeOperatorSetAVS();
+        uint256 quorumCount = _registryCoordinator.quorumCount();
+
+        address[] memory allOperators = new address[](0);
+        uint32[] memory operatorSetIdsToCreate = new uint32[](quorumCount);
+        uint32[][] memory operatorSetIds;
+
+        // Step 1: Iterate through quorum numbers and get a list of unique operators
+        for (uint8 quorumNumber = 0; quorumNumber < quorumCount; quorumNumber++) {
+            // Step 2: Get operator list for quorum at current block
+            bytes32[] memory operatorIds = _registryCoordinator.indexRegistry().getOperatorListAtBlockNumber(quorumNumber, uint32(block.number));
+        
+            // Step 3: Convert to address list and maintain a sorted array of operators
+            address[] memory operators = new address[](operatorIds.length);
+            for (uint256 i = 0; i < operatorIds.length; i++) {
+                operators[i] = _registryCoordinator.blsApkRegistry().getOperatorFromPubkeyHash(operatorIds[i]);
+                // Insert into sorted array of all operators
+                allOperators = mergeSortedArrays(allOperators, operators);
+            }
+            address[] memory filteredOperators = new address[](allOperators.length);
+            uint256 count = 0;
+            for (uint256 i = 0; i < allOperators.length; i++) {
+                if (allOperators[i] != address(0)) {
+                    filteredOperators[count++] = allOperators[i];
+                }
+            }
+            // Resize array to remove empty slots
+            assembly { mstore(filteredOperators, count) }
+            allOperators = filteredOperators;
+
+            operatorSetIds = new uint32[][](allOperators.length);
+            // Loop through each unique operator to get the quorums they are registered for
+            for (uint256 i = 0; i < allOperators.length; i++) {
```

Comment:
This step can occur outside the step1 loop? Since we can:

For each quorum
- Get all operators
- Insert into all operators array

After getting operators across all quorums:

For each operator in all operators
- Get quorum bitmap array

Reply:
Yeah good call!